### PR TITLE
net: ieee802154: Sort out fragmentation tests

### LINKED ISSUE #19761: tests/net/ieee802154/fragment failed on reel board.
**Describe the bug**
There is a "kernel panic" error when running tests/net/ieee802154/fragment case on reel board.

**To Reproduce**
1. west build -b=reel_board tests/net/ieee802154/fragment
2. west flash
3. See error

**Screenshots or console output**
I: nRF5 802154 radio initialized
W: You have 1 IPv6 net_if addresses but 2 network interfaces
W: Consider increasing CONFIG_NET_IF_MAX_IPV6_COUNT value.
***** Booting Zephyr OS build zephyr-v2.0.0-1126-g91384ef01eb6 *****
Running test suite ieee802154_fragment

starting test - test_fragment_sam00_dam00
PASS - test_fragment_sam00_dam00

starting test - test_fragment_sam01_dam01
ASSERTION FAIL [net_pkt_lladdr_dst(pkt)->addr] @ ZEPHYR_BASE/subsys/net/ip/6lo.c:511
E: r0/a1:  0x00000004  r1/a2:  0x000001ff  r2/a3:  0x00000001
E: r3/a4:  0x0000f784 r12/ip:  0x00000f01 r14/lr:  0x00003c51
E:  xpsr:  0x61000000
E: Faulting instruction address (r15/pc): 0x0000caf6
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic
E: Current thread: 0x20000aac (unknown)
E: Halting system

**Environment (please complete the following information):**
 - OS: Fedora28
 - Toolchain: zephyr-sdk-0.10.3
 - Commit ID: 91384ef01eb6

### PR DESCRIPTION
**Pull request by rlubos**
Fragmentation tests in `tests/net/ieee802154` failed due to asserts on LL addresses and a memory corruption issue. This appears to be happening since the recent 6lowpan rework.

@jukkar It seems that these tests are not automatically executed during the sanity-check, that's why it went so long unnoticed. A short investigation showed up that the ieee802514 tests have the following dependency: `depends_on: ieee802154`, yet no emulated qemu board sets `ieee802154` in their capabilities. IMO we could fix it in this PR as well. Shall we add the capability to any of the boards or remove the dependency? The latter should be no harm, as the tests seem to provide either dummy network interface or dummy ieee802154 driver.

Fixes #19761.

### CODE REVIEW

File: tests/net/ieee802154/fragment/src/main.c [hunk 1]
```diff
@@ -287,6 +287,17 @@ static struct net_pkt *create_pkt(struct net_fragment_data *data)
 		}
 	}
 
+	/* Setup link layer addresses. */
+	static u16_t dummy_short_addr;
```
Comment (tbursztyka) [inline tests/net/ieee802154/fragment/src/main.c]: no found of this, but up to you to change. Seems to be accepted in zephyr...

**Review by tbursztyka — APPROVED** [review]
wasn't an issue opened about it ? (or maybe was is just in a  slack talk?)

File: tests/net/ieee802154/fragment/src/main.c (hunk 1 above)
Comment (rlubos) [reply tests/net/ieee802154/fragment/src/main.c]: > no found of this, but up to you to change

I didn't catch this up, could you elaborate?

**Comment by rlubos** [comment]
> wasn't an issue opened about it ? (or maybe was is just in a slack talk?)

Actually I'm not aware of either, I just found the problem out when running on hardware tests on nRF52840.

File: tests/net/ieee802154/fragment/src/main.c (hunk 1 above)
Comment (tbursztyka) [reply tests/net/ieee802154/fragment/src/main.c]: well it seems to be accepted to declare variable in a middle of a code block.

**Comment by rlubos** [comment]
Oh, this actually could be the issue: https://github.com/zephyrproject-rtos/zephyr/issues/19761. I'll mention it in the original post.

File: tests/net/ieee802154/fragment/src/main.c (hunk 1 above)
Comment (rlubos) [reply tests/net/ieee802154/fragment/src/main.c]: Ah right, I can move it upwards, no prob

**Review by jukkar — APPROVED** [review]
LGTM. Perhaps you could add "Fixes #19761" to commit message too.

**Comment by jukkar** [comment]
> IMO we could fix it in this PR as well. Shall we add the capability to any of the boards or remove the dependency? The latter should be no harm, as the tests seem to provide either dummy network interface or dummy ieee802154 driver.

Agreed, we should make the tests run by sanitycheck automatically so I suppose we would need to add the capability to some board.

**Comment by rlubos** [comment]
According to https://github.com/zephyrproject-rtos/zephyr/commit/de8d755c5403d7ac43bfd8c4c5860abb33c1a3a4, it seems that `qemu_x86` is favoured for the networking tests, I'll extend its capabilities then.

**Review by alexanderwachter — APPROVED** [review]
Thanks for the 6lo fix :-)

**Pull request merged**